catalog/to-consul/syncer_test.go

```diff
@@ -155,6 +155,51 @@ func TestConsulSyncer_reapService(t *testing.T) {
 	}
 }
 
+// Test that the syncer doesn't reap any services until the initial sync has
+// been performed.
+func TestConsulSyncer_noReapingUntilInitialSync(t *testing.T) {
+	t.Parallel()
+
+	a, err := testutil.NewTestServerT(t)
+	require.NoError(t, err)
+	defer a.Stop()
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(t, err)
+	s, closer := testConsulSyncer(client)
```

Comment:
Suggestion:
```
s, closer := testConsulSyncerWithConfig(client, func(s *ConsulSyncer) {
		s.SyncPeriod = 5 * time.Millisecond
	})
```

Since the `testConsulSyncer` also `Runs` the syncer, the sync period change may not get picked up if it's made afterwards.

Reply:
Good catch. PR coming.

Reply:
https://github.com/hashicorp/consul-k8s/pull/227